app/src/main/java/net/squanchy/navigation/HomeActivity.java

```diff
@@ -111,28 +111,41 @@ protected void onStart() {
     }
 
     private void askProximityPersmissionToStartRadar() {
-        int permissionCheck = ContextCompat.checkSelfPermission(this,
-                Manifest.permission.ACCESS_FINE_LOCATION);
-        if (permissionCheck == PackageManager.PERMISSION_GRANTED) {
+        if (hasLocationPermission()) {
             proximityService.startRadar();
         } else {
-            ActivityCompat.requestPermissions(this,
-                    new String[]{Manifest.permission.ACCESS_FINE_LOCATION},
-                    MY_PERMISSIONS_REQUEST_ACCESS_FINE_LOCATION);
+            ActivityCompat.requestPermissions(
+                    this,
+                    new String[] { Manifest.permission.ACCESS_FINE_LOCATION },
+                    MY_PERMISSIONS_REQUEST_ACCESS_FINE_LOCATION
+            );
         }
     }
 
     @Override
     public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
-        if (requestCode == MY_PERMISSIONS_REQUEST_ACCESS_FINE_LOCATION)
-            if (grantResults.length > 0
-                    && grantResults[0] == PackageManager.PERMISSION_GRANTED){
+        if (requestCode == MY_PERMISSIONS_REQUEST_ACCESS_FINE_LOCATION) {
+            if (hasGrantedFineLocationAccess(grantResults)) {
                 proximityService.startRadar();
             }
+        }
+    }
+
+    private boolean hasLocationPermission() {
+        final int granted = ContextCompat.checkSelfPermission(
+                this,
+                Manifest.permission.ACCESS_FINE_LOCATION
+        );
+        return granted == PackageManager.PERMISSION_GRANTED;
+    }
+
+    private boolean hasGrantedFineLocationAccess(@NonNull int[] grantResults) {
```

Comment:
Everything is `@NonNull` by convention in our code, you can take that annotation out. It only makes sense to leave it when we're at the edge of 3rd party APIs (including the framework), but not when calling our own code.